### DISCUSSION TITLE
GitHub API rate limit

### DISCUSSION THREAD

**Discussion by favoyang** [openupm/openupm#85] [Ideas]
Rate limit of GitHub unauthenticated request is 60/hr (refs https://developer.github.com/v3/rate_limit/.) Adding package and browse package detail page will both consume API calls. Need find a way to improve the user experience. Perhaps by removing GitHub API request for package detail page first.

**Comment by dahlia** [comment]
According to [GitHub API docs][1]:

> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the user making requests.

How about make users to sign in their GitHub account?  If we can take an access token for each user it would be virtually limitless.

[1]: https://developer.github.com/v3/#rate-limiting

**Comment by favoyang** [comment]
That would be ideally when adding a package.
